wurst/systems/core/ClassSelection.wurst

```diff
@@ -31,18 +31,85 @@ let spawns = [
     gg_rct_revive_12
 ]
 
+function LinkedList<T>.take<T>(int numElems) returns LinkedList<T>
+    let list = new LinkedList<T>()
+    let take = min(numElems, this.size())
+    for i = 0 to take
+        list.add(this.get(i))
+    return list
+
 public class ClassSelector
     static let spawns = new HashMap<unit, thistype>()
+    static LinkedList<thistype> spawners
+
+    static function findByTotem(unit totem) returns thistype
+        return spawns.get(totem)
+
+    static function resetAllianceState()
+        for i = 0 to bj_MAX_PLAYER_SLOTS - 1
+            for j = 0 to bj_MAX_PLAYER_SLOTS - 1
+                if i != j
+                    SetPlayerAllianceStateBJ(players[i], players[j], bj_ALLIANCE_UNALLIED)
+
+    static function initializeTribes()
+        let numTribes = gameConfig.getNumTribes()
+        for tribeId = 0 to numTribes - 1
+            new Tribe(tribeId)
+
+    static function initializeSpawns()
+        let maxTribeId = Tribe
+            .getTribes()
+            .filter(tribe -> not tribe.isDefeated())
+            .peek()
+            .getId()
+
+        let spawnList = new LinkedList<rect>()
+        for i = 0 to maxTribeId
+            spawnList.push(spawnRects[i])
+
+        if gameConfig.isRandomSpawns()
+            spawnList.shuffle()
+
+        spawners = Tribe.getTribes().map() tribe ->
+            let spawn = spawnList.get(tribe.getId())
+            tribe.setSpawn(spawn)
+            return new ClassSelector(tribe)
+
+    static function startClassSelection()
+        resetAllianceState()
+        initializeTribes()
+        initializeSpawns()
+
+        var selectionTime = 0
+
+        if not gameConfig.isAllTrollEnabled()
+            selectionTime = gameConfig.getSelectionTimeAllowed()
+            print(GENERAL_COLOR + "Choose a troll class|r")
+            print(GENERAL_COLOR + "You will be appointed a random troll if you do not pick|r")
+
+        getTimer()
+        ..doAfter(selectionTime.toReal(), -> begin
```

Comment:
Why do you use this type of syntax for a lambda in this case but not a lambda block in others?

Reply:
If I want to chain calls after I'll use `begin`/`end`